51-dracut-rescue.install

```diff
@@ -33,11 +33,15 @@ dropindirs_sort()
 
 [[ -f /etc/os-release ]] && . /etc/os-release
 
-if [[ ! -f /etc/machine-id ]] || [[ ! -s /etc/machine-id ]]; then
-    systemd-machine-id-setup
+if [[ ${KERNEL_INSTALL_MACHINE_ID+x} ]]; then
```

Comment:
${FOO+x} equls to x if FOO is defined, and is blank if not defined. So, the above condition hits only if KERNEL_INSTALL_MACHINE_ID is defined.

Actually, the code can be altered without +x. But in this code, I intend to switch the case that systemd is 234+ or not.